sdk/push/src/main/java/com/klaviyo/push/KlaviyoPushService.kt

```diff
@@ -26,26 +50,82 @@ class KlaviyoPushService : FirebaseMessagingService() {
         fun getCurrentPushToken(): String {
             return KlaviyoPreferenceUtils.readStringPreference(PUSH_TOKEN_PREFERENCE_KEY) ?: ""
         }
+
+        /**
+         * Logs an $opened_push event for a remote notification that originated from Klaviyo
+         *
+         * TODO customer and event properties should both be optional, once SDK is tracking customer state internally
+         *
+         * @param pushPayload The data attributes of the push notification payload
+         * @param customerProperties Profile with which to associate the event
+         * @param eventProperties Optional additional properties for the event
+         */
+        fun handlePush(
+            pushPayload: Map<String, String>,
+            customerProperties: KlaviyoCustomerProperties,
+            eventProperties: KlaviyoEventProperties? = null
+        ) {
+            if (pushPayload["origin"] != "Klaviyo") {
+                return
+            }
+
+            val properties = eventProperties ?: KlaviyoEventProperties()
+            properties.addCustomProperty("push_token", getCurrentPushToken())
+            pushPayload.forEach { (key, value) ->
+                properties.addCustomProperty(key, value)
+            }
+            Klaviyo.track(KlaviyoEvent.OPENED_PUSH, customerProperties, properties)
+        }
+
+        /**
+         * Logs an $opened_push event for a remote notification that originated from Klaviyo
+         * After being opened from the system tray
+         *
+         * @param notificationIntent The Intent generated from tapping the notification
+         * @param customerProperties Profile with which to associate the event
+         * @param eventProperties Optional additional properties for the event
+         */
+        fun handlePush(
+            notificationIntent: Intent?,
+            customerProperties: KlaviyoCustomerProperties,
+            eventProperties: KlaviyoEventProperties? = null
+        ) {
+            if (notificationIntent is Intent) {
+                val payload = mutableMapOf<String, String>()
+
+                notificationIntent.extras?.keySet()?.forEach { key ->
+                    payload[key] = notificationIntent.getStringExtra(key) ?: ""
+                }
```

Comment:
more kotlin way to do this:
```
        notificationIntent?.extras?.let { extras ->
            val payload = extras.keySet().associateWith { key ->
                extras.getString(key, "")
            }
            handlePush(payload, customerProperties, eventProperties)
        }
```

Reply:
no need to create a mutable map or do the instance check anymore

Reply:
a little cleaner:
```
        val payload = notificationIntent?.extras?.let { extras ->
            extras.keySet().associateWith { key ->
                extras.getString(key, "")
            }
        } ?: return
        handlePush(payload, customerProperties, eventProperties)
```

Reply:
Nice